plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceService.java

```diff
@@ -212,6 +224,70 @@ public static Collection<QueryGroup> getFromClusterStateMetadata(String name, Cl
             .collect(Collectors.toList());
     }
 
+    /**
+     * Modify cluster state to delete the QueryGroup
+     * @param name - the name for QueryGroup to be deleted
+     * @param listener - ActionListener for DeleteQueryGroupResponse
+     */
+    public void deleteInClusterStateMetadata(String name, ActionListener<DeleteQueryGroupResponse> listener) {
+        clusterService.submitStateUpdateTask(SOURCE, new ClusterStateUpdateTask(Priority.NORMAL) {
+            @Override
+            public ClusterState execute(ClusterState currentState) throws Exception {
+                return deleteQueryGroupInClusterState(name, currentState);
+            }
+
+            @Override
+            public ThrottlingKey getClusterManagerThrottlingKey() {
+                return deleteQueryGroupThrottlingKey;
+            }
+
+            @Override
+            public void onFailure(String source, Exception e) {
+                logger.warn("Failed to delete QueryGroup due to error: {}, for source: {}", e.getMessage(), source);
+                listener.onFailure(e);
+            }
+
+            @Override
+            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                final Optional<QueryGroup> queryGroupToRemove = oldState.metadata()
+                    .queryGroups()
+                    .values()
+                    .stream()
+                    .filter(queryGroup -> queryGroup.getName().equals(name))
+                    .findFirst();
+
+                assert queryGroupToRemove.isPresent();
```

Comment:
Is it necessary to assert the presence, should we silently succeed or throw `ResourceNotFound` exception?